flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/delegation/ParserImpl.java

```diff
@@ -76,16 +81,42 @@ public ParserImpl(
         this.sqlExprToRexConverterFactory = sqlExprToRexConverterFactory;
     }
 
+    /**
+     * Parses a given statement to {@link Operation}. the statement should be complete and valid, or
+     * an {@link SqlParserException} would be thrown.
+     *
+     * @param statement input statement.
+     * @return parsed operations.
+     */
+    @Override
+    public List<Operation> parse(String statement) {
+        return parseInternal(statement, false);
+    }
+
+    /**
+     * Parses a given statement interactively to {@link Operation}, used in sql client. Different
+     * from {@link #parse(String)}, it returns a {@link StagingOperation} when given an incomplete
```

Comment:
Please don't use `StagingOperation`, as it's confusing the fact that this is an implementation of an `Operation`. Create an appropriate new union type that either has the `List<Operation>` in case the parsing is complete, or has only `SqlParseException` in case of failure, or some flag to say `expected <EOF>`. e.g. a type like:


`InteractiveParseResult`:
* `isParsingCompleted()` and `getParseResult()` in case parsing is complete
* `isParsingFailured()` and `getParseError()` in case parsing is complete but there was an error
* `isParsingIncomplete()` if the input string was an incomplete statement

Of course choose the names you think fit most, those are just a suggestion of how this type should look like

Reply:
hi, I have investigated again. currently we don't need the new function: when the statement is incomplete, parser throws SqlParserEOFException, and we can just check the exception in sql client. 
WDYT?